marklogic-data-hub-spark-connector/src/test/java/com/marklogic/hub/spark/sql/sources/v2/WriteDataTest.java

```diff
@@ -78,26 +69,23 @@ public void testBulkIngestWithoutUriPrefix() throws IOException {
 
     @Test
     public void ingestWithoutCustomApiWithCustomWorkunit(){
-        Map<String, String> params = getHubPropertiesAsMap();
-
-        Exception ex = Assertions.assertThrows(RuntimeException.class,
-            () -> {
-            DataSourceOptions options = configureDataSourceOptions(params,"2", "", "", "{\"userDefinedValue\":"+0+"}", "");
-            buildDataWriter(options);
-        });
-        assertTrue(ex.getMessage().equals("Cannot override default endpointState and/or workUnit. Please provide custom endpoint."));
+        ObjectNode customWorkUnit = objectMapper.createObjectNode();
+        customWorkUnit.put("userDefinedValue", 0);
+
+        RuntimeException ex = assertThrows(RuntimeException.class,
+            () -> buildDataWriter(new Options(getHubPropertiesAsMap()).withIngestWorkUnit(customWorkUnit)),
+            "Expected an error because a custom work unit was provided without a custom API path"
```

Comment:
@anu3990  @SameeraPriyathamTadikonda When verifying we catch an exception, we always want to include an assertion message stating why we expect the exception to be thrown.